src/lib.rs

```diff
@@ -117,6 +123,15 @@ impl Wal {
     {
         debug!("Wal {{ path: {:?} }}: opening", path.as_ref());
 
+        let max_closed_to_preserve = if options.max_closed_to_preserve >= 1 {
+            options.max_closed_to_preserve
+        } else {
+            return Err(Error::new(
+                ErrorKind::InvalidInput,
+                "max_closed_to_preserve must be at least 1",
+            ));
+        };
```

Comment:
We _could_ use [`NonZeroUsize`](https://doc.rust-lang.org/std/num/type.NonZeroUsize.html) here which would make this impossible